db/migrate/20241009214601_add_super_admin_to_users.rb

```diff
@@ -1,5 +1,23 @@
 class AddSuperAdminToUsers < ActiveRecord::Migration[7.2]
   def change
-    add_column :users, :super_admin, :boolean, default: false
+    reversible do |dir|
+      dir.up do
+        change_column :users, :role, :string, default: 'member'
+
+        execute <<-SQL
+          DROP TYPE user_role;
+        SQL
+      end
+
+      dir.down do
+        execute <<-SQL
+          CREATE TYPE user_role AS ENUM ('admin', 'member');
+        SQL
+
+        change_column_default :users, :role, nil
+        change_column :users, :role, :user_role, using: 'role::user_role'
+        change_column_default :users, :role, 'member'
+      end
+    end
```

Comment:
We have an existing `role` on the `User` table that I turned into a string (leverage ActiveRecord more, make more flexible) and added the `super_admin` option to the enum in the `User` model.

Reply:
My thought on the "super_admin" _column_ vs enum is that the "role" column is directly related to a user's actual Maybe account whereas "super_admin" is more of an internal function of administrating the product.

Maybe that's splitting hairs, though?

Reply:
Long term if the "super admin" starts severely diverging in capabilities within the app, I'd probably look towards creating a brand new table, `SuperAdminUser` that doesn't have a `Family`, `Account`s, or other domain related associations / properties so it's sole purpose is to log in as a "superadmin" and do administrative tasks.  But since these current super admins are logging in with regular `User` accounts, I think it makes sense that they have an "elevated privilege role" within the app's features.

Reply:
that's fair. in reality we'd likely just build an entirely separate admin UI and flow anyways as having support members be logged in to their own Maybe account is a bit of a weird flow anyways.

Reply:
Yep, agree!